librad/src/net/protocol/membership/periodic.rs

```diff
@@ -56,7 +57,17 @@ where
         }
     });
 
-    if let Err(e) = stream::select(shuffle, promote).map(Ok).forward(tx).await {
+    let tickle = Interval::new(Duration::from_secs(15), Duration::from_secs(5))
+        .filter_map(|_| future::ready(Some(Periodic::Tickle)));
+
+    // FIXME(xla): There should be a more elegant way to merge these streams into
```

Comment:
No I don't think there's a better way

Reply:
Using another language could be a good option :upside_down_face:

Reply:
Consumer! Just get your hands dirty on implementing RFC 424242: Statically Dispatchable Existentially Quantified Types already!

Reply:
#sauvage